packages/task/src/browser/task-configurations.ts

```diff
@@ -250,7 +250,7 @@ export class TaskConfigurations implements Disposable {
             const taskDefinition = this.taskDefinitionRegistry.getDefinition(taskConfig);
             if (taskDefinition) {
                 const cus = customizationByType.filter(customization =>
-                    taskDefinition.properties.required.every(rp => customization[rp] === taskConfig[rp])
+                    taskDefinition.properties.required?.every(rp => customization[rp] === taskConfig[rp])
```

Comment:
If no `required` is specified, should this pass or fail? This code will filter out anything where `required` is not defined, but if having no required is the same as having `[]`, then the `every` check would pass.

Could do `(taskDefinition.properties.required ?? [])...` or `taskDefinition.properties.required?.every(...) ?? true`

Reply:
Ah because `[].every(...)` returns `true`, makes sense.